spec.html

```diff
@@ -4078,36 +4078,36 @@ <h1>Type Conversion</h1>
     <p>The BigInt type has no implicit conversions in the ECMAScript language; programmers must call BigInt explicitly to convert values from other types.</p>
 
     <emu-clause id="sec-toprimitive" aoid="ToPrimitive" oldids="table-9">
-      <h1>ToPrimitive ( _input_ [ , _PreferredType_ ] )</h1>
-      <p>The abstract operation ToPrimitive takes argument _input_ and optional argument _PreferredType_. It converts its _input_ argument to a non-Object type. If an object is capable of converting to more than one primitive type, it may use the optional hint _PreferredType_ to favour that type. It performs the following steps when called:</p>
+      <h1>ToPrimitive ( _input_ [ , _preferredType_ ] )</h1>
+      <p>The abstract operation ToPrimitive takes argument _input_ and optional argument _preferredType_. It converts its _input_ argument to a non-Object type. If an object is capable of converting to more than one primitive type, it may use the optional hint _preferredType_ to favour that type. It performs the following steps when called:</p>
       <emu-alg>
         1. Assert: _input_ is an ECMAScript language value.
         1. If Type(_input_) is Object, then
-          1. If _PreferredType_ is not present, let _hint_ be *"default"*.
-          1. Else if _PreferredType_ is hint String, let _hint_ be *"string"*.
-          1. Else,
-            1. Assert: _PreferredType_ is hint Number.
-            1. Let _hint_ be *"number"*.
           1. Let _exoticToPrim_ be ? GetMethod(_input_, @@toPrimitive).
           1. If _exoticToPrim_ is not *undefined*, then
+            1. If _preferredType_ is not present, let _hint_ be *"default"*.
+            1. Else if _preferredType_ is ~string~, let _hint_ be *"string"*.
+            1. Else,
+              1. Assert: _preferredType_ is ~number~.
+              1. Let _hint_ be *"number"*.
```

Comment:
kind of seems like it'd be simpler to just pass the string instead of doing a conversion from "not a string" to "basically the equivalent string"

Reply:
Eh, I'd weakly prefer to pass the spec enum at all the call sites and create a string here. If we used a string, it'd be serving a dual purpose: signalling the desired behaviour of OrdinaryToPrimitive and passing to user functions. Having two separate values (even if isomorphic) seems like it separates those concepts better.

Reply:
Fair enough.